orangecanvas/scheme/readwrite.py

```diff
@@ -801,6 +801,9 @@ def literal_dumps(obj, indent=None, relaxed_types=True):
     builtins_mapping = {dict}
 
     def check(obj):
+        if type(obj) == float and obj != obj:
+            raise TypeError("NaN can not be serialized as a python literal")
```

Comment:
```suggestion
        if type(obj) == float and not math.isfinite(obj):
            raise TypeError("Non-finite values can not be serialized as a python literal")
```
inf values have the same problem.